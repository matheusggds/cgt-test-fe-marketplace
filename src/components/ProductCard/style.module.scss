.ProductCard {
  display: flex;
  flex-direction: column;
  height: 100%;
}

.ProductContent {
  flex: 1;
}
