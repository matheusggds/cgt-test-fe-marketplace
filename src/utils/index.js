// convert to dolar string
export const formatToCurrency = (val) =>
  Intl.NumberFormat("en-US", {
    style: "currency",
    currency: "USD",
  }).format(val);
