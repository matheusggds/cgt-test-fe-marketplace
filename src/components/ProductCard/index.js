import {
  Button,
  Card,
  CardActionArea,
  CardActions,
  CardContent,
  CardMedia,
  Typography,
} from "@mui/material";

import { Link } from "react-router-dom";

import React from "react";

import style from "./style.module.scss";
import { formatToCurrency } from "../../utils";
import useCartContext from "../../context/cart";

const ProductCard = ({ data }) => {
  const { price, thumbnail, description, title, id } = data;
  const [_, actions] = useCartContext();

  return (
    <Card className={style.ProductCard}>
      <CardActionArea
        sx={{ height: "100%" }}
        component={Link}
        to={"/product/" + id}
      >
        <CardMedia component="img" height="140" image={thumbnail} alt={title} />
        <CardContent className={style.ProductContent}>
          <Typography gutterBottom variant="h5" component="div">
            {title}
          </Typography>
          <Typography variant="body2" color="text.secondary">
            {description}
          </Typography>
          <Typography sx={{ mt: 3, align: "center" }}>
            Price: <b>{formatToCurrency(price)}</b>
          </Typography>
        </CardContent>
      </CardActionArea>
      <CardActions>
        <Button
          onClick={() => actions.addItem(data, 1)}
          size="small"
          color="primary"
          variant="outlined"
        >
          add to cart
        </Button>
      </CardActions>
    </Card>
  );
};

export default ProductCard;
