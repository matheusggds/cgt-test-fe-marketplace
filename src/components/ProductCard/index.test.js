import { render, screen, waitFor } from "@testing-library/react";
import { renderWithContext } from "../../utils";
import ProductCard from "./";

const MOCK_CARD = {
  id: 1,
  price: 5,
  title: "test title",
  description: "test description",
  thumbnail: "test/path/file.jpg",
};

describe("<ProductCart />", () => {
  describe("View", () => {
    it("should display default layout", async () => {
      const productCard = renderWithContext(<ProductCard data={MOCK_CARD} />);

      expect(productCard).toMatchSnapshot();
    });
  });
});
