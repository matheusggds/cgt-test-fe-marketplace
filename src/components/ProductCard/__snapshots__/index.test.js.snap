// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductCart /> View should display default layout 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root ProductCard css-bhp9pd-MuiPaper-root-MuiCard-root"
      >
        <a
          class="MuiButtonBase-root MuiCardActionArea-root css-1pz8km4-MuiButtonBase-root-MuiCardActionArea-root"
          href="/product/1"
          tabindex="0"
        >
          <img
            alt="test title"
            class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
            height="140"
            src="test/path/file.jpg"
          />
          <div
            class="MuiCardContent-root ProductContent css-46bh2p-MuiCardContent-root"
          >
            <div
              class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-h93ljk-MuiTypography-root"
            >
              test title
            </div>
            <p
              class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
            >
              test description
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1 css-1yoa5d8-MuiTypography-root"
            >
              Price: 
              <b>
                $5.00
              </b>
            </p>
          </div>
          <span
            class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
          />
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </a>
        <div
          class="MuiCardActions-root MuiCardActions-spacing css-1t6e9jv-MuiCardActions-root"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall css-1k23hlb-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            add to cart
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root ProductCard css-bhp9pd-MuiPaper-root-MuiCard-root"
    >
      <a
        class="MuiButtonBase-root MuiCardActionArea-root css-1pz8km4-MuiButtonBase-root-MuiCardActionArea-root"
        href="/product/1"
        tabindex="0"
      >
        <img
          alt="test title"
          class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
          height="140"
          src="test/path/file.jpg"
        />
        <div
          class="MuiCardContent-root ProductContent css-46bh2p-MuiCardContent-root"
        >
          <div
            class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-h93ljk-MuiTypography-root"
          >
            test title
          </div>
          <p
            class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
          >
            test description
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 css-1yoa5d8-MuiTypography-root"
          >
            Price: 
            <b>
              $5.00
            </b>
          </p>
        </div>
        <span
          class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
        />
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </a>
      <div
        class="MuiCardActions-root MuiCardActions-spacing css-1t6e9jv-MuiCardActions-root"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall css-1k23hlb-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          add to cart
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
